marshal/proto_test.go

```diff
@@ -0,0 +1,142 @@
+package marshal
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xitongsys/parquet-go/schema"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type subElem struct {
+	Val string
+}
+
+type testNestedElem struct {
+	SubElem     subElem
+	SubPtr      *subElem
+	SubList     []subElem
+	SubRepeated []*subElem
+	// Marshal ok but not write since it doesn't have corresponding primitive type
+	EmptyElem     struct{}
+	EmptyPtr      *struct{}
+	EmptyList     []struct{}
+	EmptyRepeated []*struct{}
+}
+type JobStatus int32
+
+const (
+	JobStatus_JobStatus_UNSPECIFIED  JobStatus = 0
+	JobStatus_BLOCKED                JobStatus = 1
+	JobStatus_ENQUEUED               JobStatus = 2
+	JobStatus_RUNNING                JobStatus = 3
+	JobStatus_COMPLETED              JobStatus = 4
+	JobStatus_ERRORED                JobStatus = 5
+	JobStatus_CANCELLED              JobStatus = 6
+	JobStatus_UPSTREAM_NOT_PROCESSED JobStatus = 7
+)
+
+func (x JobStatus) Enum() *JobStatus {
+	p := new(JobStatus)
+	*p = x
+	return p
+}
+
+func (x JobStatus) String() string {
+	statusToName := map[int32]string{
+		0: "ASYNCJOBSTATUS_UNSPECIFIED",
```

Comment:
Any reason this one's naming convention is different than the other items? Why not `UNSPECIFIED` to keep it consistent with everything else?

Reply:
This is auto generated from the proto. I copies the generated code here to test it out. I could make change but that will be different as regular proto generated code.